rllib/evaluation/sampler.py

```diff
@@ -820,6 +820,10 @@ def _process_observations(
     for env_id, all_agents_obs in unfiltered_obs.items():
         is_new_episode: bool = env_id not in active_episodes
         episode: Episode = active_episodes[env_id]
+        episode_faulty = any(
+            isinstance(i, dict) and "episode_faulty" in i
+            for i in infos[env_id].values()
+        )
```

Comment:
when would infos not be a dict?
to be completely correct, you would need:
```
isinstance(i, dict) and "episode_faulty" in i and i["episode_faulty"]
```
why do we need to aggregate episode_faulty bits actually. shouldn't we just ignore the episode from specifically faulty envs?
thinking about this a bit more, maybe we just need to mark the corresponding Episode object faulty. And the bit will carry to all the places, and be handled properly?

Reply:
Fixed the dict check (not needed anymore) and added a comment to explain that it would require us to change the BaseEnv API if we wanted to include sub-environment errors into the normal (obs, reward, etc..) return values (of the other healthy sub-envs).